src/Okta.Sdk/Api/OAuthApi.cs

```diff
@@ -183,10 +195,31 @@ public async Task<ApiResponse<OAuthTokenResponse>> GetBearerTokenWithHttpInfoAsy
             {
                 localVarRequestOptions.HeaderParameters.Add("Accept", localVarAccept);
             }
-
-            // make the HTTP request
+            
+            var dpopJwt = _dpopProofJwtGenerator.GenerateJWT();
+            localVarRequestOptions.HeaderParameters.Add("DPoP", dpopJwt);
+            
             var localVarResponse = await this.AsynchronousClient.PostAsync<OAuthTokenResponse>(accessTokenUri.Trim(), localVarRequestOptions, this.Configuration, cancellationToken).ConfigureAwait(false);
 
+            if (localVarResponse.StatusCode == HttpStatusCode.BadRequest && localVarResponse.Data.Error == "use_dpop_nonce")
+            {
+                if (localVarResponse.Headers.TryGetValue("dpop-nonce", out var header))
+                {
+                    
+                    var nonce = header.FirstOrDefault();
+                    dpopJwt = _dpopProofJwtGenerator.GenerateJWT(nonce);
+                    jwtSecurityToken = _jwtGenerator.GenerateSignedJWT();
+                    
+                    localVarRequestOptions.FormParameters.Remove("client_assertion");
+                    localVarRequestOptions.FormParameters.Add("client_assertion", jwtSecurityToken.ToString());
+
+                    localVarRequestOptions.HeaderParameters["DPoP"].Clear();
+                    localVarRequestOptions.HeaderParameters["DPoP"].Add(dpopJwt);
+
+                    localVarResponse = await this.AsynchronousClient.PostAsync<OAuthTokenResponse>(accessTokenUri.Trim(), localVarRequestOptions, this.Configuration, cancellationToken).ConfigureAwait(false);
+                }
```

Comment:
Both interfaces require slightly different things, for example, there DPoP interface requires a `RotateKeys` method because the client is responsible for keys, whereas the other JWT interface is for OAuth tokens where the consumer is responsible for keys and not RotateKeys method is required. This is why I decided to create a separate interface.

Reply:
The names of the interfaces imply a common surface, the details you describe should be expressed at the implementation level.

Reply:
Yeah, I actually had the same feeling at the beginning but decided to create a new interface to reduce the number of breaking changes for the existing interface consumers. I'll revisit it, and let's talk about this too.